packages/hash/design-system/src/use-scroll-lock.ts

```diff
@@ -3,54 +3,59 @@ import { useEffect, useLayoutEffect, useRef, useState } from "react";
 /**
  * @see https://github.com/mui/material-ui/blob/master/packages/mui-utils/src/getScrollbarSize.ts
  */
-const getScrollbarSize = (doc: Document) => {
-  const documentWidth = doc.documentElement.clientWidth;
+const getScrollbarSizeOfDocument = () => {
+  const documentWidth = document.documentElement.clientWidth;
   return Math.abs(window.innerWidth - documentWidth);
 };
 
-const useLastScrollbarSize = () => {
+const useLastScrollbarSize = (element?: HTMLElement) => {
   const [lastScrollbarSize, setLastScrollbarSize] = useState(0);
   const observerRef = useRef<ResizeObserver>();
 
   useEffect(() => {
     observerRef.current = new ResizeObserver(() => {
-      const scrollbarSize = getScrollbarSize(document);
+      const scrollbarSize = element
+        ? element.offsetWidth - element.clientWidth
+        : getScrollbarSizeOfDocument();
 
       if (scrollbarSize > 0) {
         setLastScrollbarSize(scrollbarSize);
       }
     });
 
-    observerRef.current.observe(document.body);
+    observerRef.current.observe(element ?? document.body);
 
     return () => {
       observerRef.current?.disconnect();
     };
-  }, []);
+  }, [element]);
 
   return lastScrollbarSize;
 };
 
-const removeBodyStyles = () => {
-  document.body.style.removeProperty("overflow");
-  document.body.style.removeProperty("padding-right");
+const removeStylesFromElement = (element: HTMLElement) => {
+  element.style.removeProperty("overflow");
+  element.style.removeProperty("padding-right");
 };
 
 /**
  * This function does the same thing as MUI's scroll-lock mechanism, but in a hook.
  * So we can use the same scroll-lock at custom components
+ * @param active is locked
+ * @param elementToLock an HTML element to lock it's scroll. Locks `document.body` if it's left empty
  */
-export const useScrollLock = (active: boolean) => {
-  const scrollbarSize = useLastScrollbarSize();
+export const useScrollLock = (active: boolean, elementToLock?: HTMLElement) => {
```

Comment:
You should probably make this take a ref, not the element itself, because reading the dom (as the consumer of this does with `getElementsByTagName`) can be a side effect so you probably ought not to be doing it at render time.

Reply:
Makes sense, and updated it here https://github.com/hashintel/hash/pull/1751/commits/e0ebf7eb18b95df6cb0efa2550d3a92475a642b0 (reused `EditBarContext` even if it's not related with edit bar, because you already captured the ref of `main` element there)